docs/rules/jsx-no-useless-fragment.md

```diff
@@ -0,0 +1,50 @@
+# Disallow unnecessary fragments (react/jsx-no-useless-fragment)
+
+A fragment is redundant if it contains only one child, or if it is the child of a html element.
```

Comment:
Seems like we need to also add prose and examples and tests showing that a fragment is not useless if it has a `key` prop and also appears in a context where a key prop makes sense (ie, inside an array literal or returned from a function)

Reply:
I pushed a version just let pass any keyed fragment.

Reply:
If a fragment has only a single JSXElement child, it is still useless even if it does have a key.

For example:

```
<React.Fragment key={someVar}>
  <div />
</React.Fragment>
```
Should be refactored/fixed to:

```
<div key={someVar} />
```

Reply:
That's not entirely true; something might be cloning the element.